jekyll/_cci2/deploy-android-applications.adoc

```diff
@@ -0,0 +1,333 @@
+    }
+
+    return versionName
+}
+```
++
+In this snippet, you added variables that hold your app version values. You then added two methods, `generateVersionCode` and `generateVersionName` to generate the version code and version name based on how the app version value changes. This helps to give your app a unique and progressive way of generating your version code when you modify your app versions.
++
+Note that you will need to update at least one parameter of the version for every deployment. Fastlane fails if a version code is reused. 
+
+. Update these properties in the `defaultConfig` section of the `build.gradle` file like so:
+
+```
+defaultConfig {
+    versionName generateVersionName()
+    versionCode generateVersionCode()
+    // ... Leave others as is
+
+}
+```
+
+Now your android app is able to be bundled and deployed on your local machine. 
+
+== 3. Set up CircleCI Deployment
+
+. You need to convert your keystore** to base64 to safely access it in CircleCI. You can do this conversion in the terminal using the following command: 
++
+```shell
+$ base64 your_key_store
+```
++
+Save the output somewhere easily accessible for the next step.
+
+. Next, you need to set xref:env-vars.md[environment variables] for deployment through CircleCI.
++
+Open your Android project in the link:app.circleci.com[CircleCI web app] and select *Project Settings*. Navigate to *Environment Variables* and add the following variables:
++
+- BASE64_KEYSTORE - Your base64 keystore, generated in the previous step
++
+- GOOGLE_PLAY_KEY - The contents of your `api.json` file, generated from the Fastlane install before starting this tutorial
++
+- RELEASE_KEY_ALIAS - Your key alias
++
+- RELEASE_KEY_PASSWORD - Your key password
++
+- RELEASE_STORE_PASSWORD - Your keystore password
+
+. Add the following snippet to your `build.gradle` file. This allows you to import your keystore properties from a `keystore.properties` that will be generated at runtime.
++
+```
+def keyStorePropertiesFile = rootProject.file("keystore.properties")
+def keyStoreProperties = new Properties()
+keyStoreProperties.load(new FileInputStream(keyStorePropertiesFile))
+
+android {
+...
+}
+```
++
+[NOTE]
+====
+You can also create a `keystore.properties` file in your project directory for local bundling and deployment, if you wish. Do *not* push this file or your keystore to a remote repository. 
+
+Use the following snippet to create this file:
+
+```
+releaseKeyAlias=YourKeyAlias
+releaseKeyPassword=YourKeyPassword
+releaseKeyStore=YourKeyStorePath
+releaseStorePassword=YourKeyStorePassword
+```
+====
+
+Now, you need to configure `.circleci/config.yml` to decrypt your keystore, generate `keystore.properties`, and create the Google Play API key at runtime.
+
+If you have not already done so, create a `.circleci` folder in the root of your project repository. Inside the `.circleci` folder, create a `config.yml` file.
+
+== 4.A. Set up config with the Android orb
```

Comment:
If this is in the style guide then definitely do that, but I might reformat this as 4a. (and 4b.)

4.A. just kind of looks a little weird to me.